backend/test/integration/defenceController.test.ts

```diff
@@ -0,0 +1,312 @@
+			const req = {
+				body: {
+					defenceId: DEFENCE_ID.CHARACTER_LIMIT,
+					level,
+					configItemId: 'MAX_MESSAGE_LENGTH',
+				},
+				session: {
+					levelState: getInitialLevelStates(),
+				},
+			} as DefenceConfigItemResetRequest;
+
+			const res = responseMock();
+
+			handleResetDefenceConfigItem(req, res);
+
+			expect(res.status).toHaveBeenCalledWith(400);
+			expect(res.send).toHaveBeenCalledWith(
+				'You cannot reset defence config items on this level, because it uses the default defences'
+			);
+		});
+
+		test(`GIVEN level ${
+			level + 1
+		} WHEN attempt to get defence status THEN shoult return undefined`, () => {
+			const req = {
+				query: { level },
+				session: {
+					levelState: getInitialLevelStates(),
+				},
+			} as unknown as DefenceStatusRequest;
+
+			const res = responseMock();
+
+			handleGetDefenceStatus(req, res);
+
+			expect(res.send).toHaveBeenCalledWith(undefined);
+		});
+	});
+
+	[LEVEL_NAMES.LEVEL_3, LEVEL_NAMES.SANDBOX].forEach((level) => {
+		test(`GIVEN level ${level} WHEN attempt to activate a defence THEN defence is activated`, () => {
+			const req = {
+				body: {
+					defenceId: DEFENCE_ID.CHARACTER_LIMIT,
+					level,
+				},
+				session: {
+					levelState: getInitialLevelStates(),
+				},
+			} as unknown as DefenceActivateRequest;
+
+			const res = responseMock();
+
+			handleDefenceActivation(req, res);
+
+			const newLevelState = getInitialLevelStates().map((levelState) =>
+				levelState.level === level
+					? {
+							...levelState,
+							defences: levelState.defences?.map((defence) =>
+								defence.id === DEFENCE_ID.CHARACTER_LIMIT
+									? { ...defence, isActive: true }
+									: defence
+							),
+					  }
+					: levelState
+			);
+
+			expect(res.status).toHaveBeenCalledWith(200);
+			expect(res.send).toHaveBeenCalled();
+			expect(req.session.levelState).toEqual(newLevelState);
+		});
+
+		test(`GIVEN level ${level} WHEN attempt to deactivate a defence THEN defence is deactivated`, () => {
+			const initialLevelStatesButWithCharacterLimitActive =
+				getInitialLevelStates().map((levelState) =>
+					levelState.level === level
+						? {
+								...levelState,
+								defences: levelState.defences?.map((defence) =>
+									defence.id === DEFENCE_ID.CHARACTER_LIMIT
+										? { ...defence, isActive: true }
+										: defence
+								),
+						  }
+						: levelState
+				);
+
+			const req = {
+				body: {
+					defenceId: DEFENCE_ID.CHARACTER_LIMIT,
+					level,
+				},
+				session: {
+					levelState: initialLevelStatesButWithCharacterLimitActive,
+				},
+			} as unknown as DefenceActivateRequest;
+
+			const res = responseMock();
+
+			handleDefenceDeactivation(req, res);
+
+			expect(res.status).toHaveBeenCalledWith(200);
+			expect(res.send).toHaveBeenCalled();
+			expect(req.session.levelState).toEqual(getInitialLevelStates());
+		});
+
+		test(`GIVEN level ${level} WHEN attempt to configure a defence THEN defence is configured`, () => {
```

Comment:
I don't think so - we can only configure defences in SANDBOX. Same for reset. We can activate and deactivate in Level 3 though.

If this test passes, then I'd say your implementation logic is incorrect.

Reply:
I was not going to bother locking down the API against configuring defences for level 3 for the sake of simplicity. I'll change that

Reply:
done